.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseMergedDateTimeExtractor.cs

```diff
@@ -223,7 +223,7 @@ private void AddTo(List<ExtractResult> dst, List<ExtractResult> src, string text
                 var firstIndex = -1;
                 for (var i = 0; i < dst.Count; i++)
                 {
-                    if (dst[i].IsOverlap(result))
+                    if (dst[i].IsOverlap(result) && !(dst[i].Text == result.Text && this.config.UnspecificDatePeriodRegex.IsMatch(result.Text)))
```

Comment:
Can you add a comment explaining this condition?

Reply:
the condition avoids merging results that match UnspecificDatePeriodRegex. For example in "first week", "week" is extracted by DatePeriod and by Duration and we want to keep both results separate (the DatePeriod result will be later filtered). 
Comment added.